src/ros_filter_bias_estimator.cpp

```diff
@@ -0,0 +1,261 @@
+                            //  new = clampRotation(0.8*clampRotation(350) - 175)
+                            //  new = clampRotation(0.8*-10 - 175)
+                            //  new = clampRotation(-8 - 175)
+                            //  new = clampRotation(-183)
+                            //  new = 177
+                            orientation_offset_[axis] =
+                                FilterUtilities::clampRotation(alpha_ * FilterUtilities::clampRotation(orientation_offset_[axis] - offset) + offset);
+
+                            // Should be added (+) since these are variances.
+                            // Note that the variance is not an actual angle, so angle wrapping is not required.
+                            orientation_offset_variance_[axis] = alpha_ * orientation_offset_variance_[axis] +
+                                (1.0 - alpha_) * (uncorrected_orientation_variance_[axis] + measurement_variance[axis]);
+                        }
+                        // Calculate the difference between the two filters estimates
+                        double abs_filter_difference = ::fabs(uncorrected_orientation_estimate_[axis] - orientation_estimate[axis]);
+                        // Dropped below limit or calculated once with no limit
+                        orientation_offset_has_been_set_[axis] |= (abs_filter_difference < max_divergence_) | (max_divergence_ < 1e-9);
+
+                        // Handle the divergence test initialization
+                        if(was_updating == false) {
+                            // Calculate the time
+                            // Determine the time to drop below the cutoff based on the current difference between the uncorrected and corrected estimators
+                            //  and the alpha-beta filter
+                            // The calculation is from max_divergence_ = abs_filter_difference * alpha_^(divergence_test_steps_)
+                            // divergence_test_steps_ = ln(max_divergence_ / abs_filter_difference) / ln(alpha_)
+                            // Note that max_divergence_ must be < abs_filter_difference since alpha_ < 1.0
+                            if((max_divergence_ < abs_filter_difference) && (max_divergence_ > 0.0))
+                            {
+                                divergence_test_steps_[axis] = round(log(max_divergence_ / abs_filter_difference) / log(alpha_));
+                            }
+                            else
+                            {
+                                divergence_test_steps_[axis] = 0;  // Already within tolerance
+                            }
+                            RF_TOOLS_VERBOSE("Required number of steps for axis " << axis_name <<
+                                " to test divergence: " << divergence_test_steps_[axis] << " given abs filter difference " << abs_filter_difference
+                                << " and max difference " << max_divergence_ << "\n");
```

Comment:
Is that a method  for tuning the parameter in `monitor_imu` ?

Reply:
No, I don't think so. It works here because of how this filter works, but the monitor_imu parameter and filter works differently. I'll think about it, though, and see if there might be a way to use something similar.